pennylane/plugins/expt_tensornet.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Experimental simulator plugin based on tensor network contractions
+"""
+import functools
+
+import numpy as np
+import tensornetwork as tn
+
+from pennylane.utils import _flatten
+from pennylane import Device
+from pennylane.plugins.default_qubit import I, X, Y, Z, H, CNOT, SWAP, CZ, S, T, CSWAP, \
+    Rphi, Rotx, Roty, Rotz, Rot3, CRot3, CRotx, CRoty, CRotz, CRot3, hermitian, identity, \
+    unitary
+
+# tolerance for numerical errors
+tolerance = 1e-10
+
+#========================================================
+#  device
+#========================================================
+
+
+class TensorNetwork(Device):
+    """Experimental Tensor Network simulator device for PennyLane.
+
+    Args:
+        wires (int): the number of modes to initialize the device in
+    """
+    name = 'PennyLane TensorNetwork simulator plugin'
+    short_name = 'expt.tensornet'
+    pennylane_requires = '0.7'
+    version = '0.7.0'
+    author = 'Xanadu Inc.'
+    _capabilities = {"model": "qubit", "tensor_observables": True}
+
+    _operation_map = {
+        'QubitStateVector': None,
+        'BasisState': None,
+        'QubitUnitary': unitary,
+        'PauliX': X,
+        'PauliY': Y,
+        'PauliZ': Z,
+        'Hadamard': H,
+        'S': S,
+        'T': T,
+        'CNOT': CNOT,
+        'SWAP': SWAP,
+        'CSWAP':CSWAP,
+        'CZ': CZ,
+        'PhaseShift': Rphi,
+        'RX': Rotx,
+        'RY': Roty,
+        'RZ': Rotz,
+        'Rot': Rot3,
+        'CRX': CRotx,
+        'CRY': CRoty,
+        'CRZ': CRotz,
+        'CRot': CRot3
+    }
+
+    _observable_map = {
+        'PauliX': X,
+        'PauliY': Y,
+        'PauliZ': Z,
+        'Hadamard': H,
+        'Hermitian': hermitian,
+        'Identity': identity
+    }
+
+    def __init__(self, wires):
+        super().__init__(wires, shots=1)
+        self.eng = None
+        self.analytic = True
+        self._nodes = []
+        self._edges = []
+        self._zero_state = np.zeros([2] * wires)
+        self._zero_state[tuple([0] * wires)] = 1.0
```

Comment:
Just out of curiosity, does this raises a deprecation warning? I have a feeling NumPy is changing to requiring the indexing tuple be an array

Reply:
I think so, yes

Reply:
Might as well change it to an array now then, rather than having to do it later

Reply:
Also, once NumPy does make the plunge and deprecates this, I think it will _really_ break autograd :laughing:

Reply:
Actually, I just checked and it doesn't give an error. Also, using an array instead of a tuple leads to different behaviour (i.e., more than one element is set to 1)

Reply:
tested on numpy 1.17.3

Reply:
Cool, I probably had the deprecation backwards (they used to do the same thing, now they have different behaviour)